apps/chat/agent/tools.py

```diff
@@ -125,21 +123,28 @@ def _move_datetime_to_new_weekday_and_time(date: datetime, new_weekday: int, new
     return date.replace(hour=new_hour, minute=new_minute, second=0) + timedelta(days=day_diff)
 
 
-def create_periodic_task(experiment_session: ExperimentSession, message: str, **kwargs):
-    task_kwargs = json.dumps(
-        {
-            "chat_ids": [experiment_session.participant.identifier],
-            "message": message,
-            "is_bot_instruction": False,
-            "experiment_public_id": str(experiment_session.experiment.public_id),
-        }
-    )
-    PeriodicTask.objects.create(
-        name=f"reminder-{experiment_session.id}-{uuid.uuid4()}",
-        task=BOT_MESSAGE_FOR_USER_TASK,
-        kwargs=task_kwargs,
-        **kwargs,
-    )
+def create_schedule_message(experiment_session: ExperimentSession, message: str, **kwargs):
+    form = ScheduledMessageConfigForm(data=kwargs, experiment_id=experiment_session.experiment.public_id)
+    if form.is_valid():
```

Comment:
should this function return errors if it's not valid

Reply:
Since the return value of this will be fed back to the agent, I think we should catch any error and return something like "Could not do that". The LLM will use this to tell the user that it couldn't create a scheduled message

Reply:
is this better? [e3cc6b2](https://github.com/dimagi/open-chat-studio/pull/527/commits/e3cc6b2779fcef5bf56c86ee140ecc359b3516c5)